action_server/src/action_server/actions/example_action.py

```diff
@@ -0,0 +1,129 @@
+from action import Action, ConfigurationData
+from robot_skills.util.entity import Entity
+
+
+class ExampleAction(Action):
+    """
+    The ExampleAction class explains how Actions, as handled by the TaskManager, work.
+    """
+    def __init__(self):
+        # Call the base class constructor
+        Action.__init__(self)
+        """
+        The base class constructor initializes the following fields:
+         - self._config_result = ConfigurationResult()
+         - self._execute_result = ActionResult()
+        Their fields should be assigned meaningful values for the caller to give decent feedback to the user.
+
+        Here, you can specify which robot skills are required to perform this action, you can specify what fields are
+        required in the semantics of the task, and what information from previous tasks is required.
+        """
+
+        """
+        You can set the required robot skills here. During the configure step, the given robot object is checked for
+        their availability.
+        """
+        self._required_skills = ['head', 'hmi', 'base']
+
+        """
+        Similarly, you can set the required keys in the knowledge. Their availability is also checked in the configure
+        step.
+        """
+        self._required_passed_knowledge = []
+
+        """
+        Similarly, you can set the required keys in the semantics. Their availability is also checked in the configure
+        step. However, a missing field means that the client may want to add this information by posing a follow-up
+        question. Therefore, this is a dict mapping from the required fields to the prompts requesting the user for
+        the specific missing information.
+
+        If information is missing, the json path to the missing field is returned to the client, so the client knows
+        which field to add.
+        """
+        self._required_field_prompts = {'object': " What object would you like me to manipulate? "}
+
+        """
+        Objects that should be passed on to a next action should be instantiated here.
+        """
+        self._object = Entity()
+
+    def _configure(self, robot, config):
+        """
+        :param robot: The robot to use for this action
+        :param config: The configuration of the action, containing fields semantics and knowledge
+        :return: ConfigurationResult
+
+        The configure step is typically done just after the command is given. It is intended to check if the task makes
+        any sense. For example, a task 'Take it to the bridge.', does not make sense as 'it' should refer to something.
+        In another example, 'Find Ava', we may want to search differently than when the task is to find an object. All
+        such checking can be done in the configure step.
+
+        Be careful when querying the world model here, because information may be added to that during execution of
+        earlier actions, which is done later.
```

Comment:
May be confusing. For a n00b, how are earlier actions going to be executed later. 
Perhaps change to  ```information about the world may be added in actions prior to this one. Those prior actions are also executed *after* all actions are configured```